tls/extensions/s2n_client_ems.c

```diff
@@ -64,6 +64,10 @@ static int s2n_client_ems_recv(struct s2n_connection *conn, struct s2n_stuffer *
  **/
 static bool s2n_client_ems_should_send(struct s2n_connection *conn)
 {
+    if (!conn) {
+        return false;
+    }
+
     /* Don't send this extension if the previous session did not negotiate EMS */
     if (conn->set_session && !conn->ems_negotiated) {
```

Comment:
There's no reason to specifically interpret a null as "don't do ems", so we could just do:
```suggestion
    /* Don't send this extension if the previous session did not negotiate EMS */
    if (conn && conn->set_session && !conn->ems_negotiated) {
```